// es-abstract/es2017.js
'use strict';

/* eslint global-require: 0 */
// https://262.ecma-international.org/8.0/#sec-abstract-operations
var ES2017 = {
	'Abstract Equality Comparison': require('es-abstract/2017/AbstractEqualityComparison'),
	'Abstract Relational Comparison': require('es-abstract/2017/AbstractRelationalComparison'),
	'Strict Equality Comparison': require('es-abstract/2017/StrictEqualityComparison'),
	abs: require('es-abstract/2017/abs'),
	AdvanceStringIndex: require('es-abstract/2017/AdvanceStringIndex'),
	ArrayCreate: require('es-abstract/2017/ArrayCreate'),
	ArraySetLength: require('es-abstract/2017/ArraySetLength'),
	ArraySpeciesCreate: require('es-abstract/2017/ArraySpeciesCreate'),
	Call: require('es-abstract/2017/Call'),
	Canonicalize: require('es-abstract/2017/Canonicalize'),
	CanonicalNumericIndexString: require('es-abstract/2017/CanonicalNumericIndexString'),
	CharacterRange: require('es-abstract/2017/CharacterRange'),
	CompletePropertyDescriptor: require('es-abstract/2017/CompletePropertyDescriptor'),
	CompletionRecord: require('es-abstract/2017/CompletionRecord'),
	CreateDataProperty: require('es-abstract/2017/CreateDataProperty'),
	CreateDataPropertyOrThrow: require('es-abstract/2017/CreateDataPropertyOrThrow'),
	CreateHTML: require('es-abstract/2017/CreateHTML'),
	CreateIterResultObject: require('es-abstract/2017/CreateIterResultObject'),
	CreateListFromArrayLike: require('es-abstract/2017/CreateListFromArrayLike'),
	CreateMethodProperty: require('es-abstract/2017/CreateMethodProperty'),
	DateFromTime: require('es-abstract/2017/DateFromTime'),
	Day: require('es-abstract/2017/Day'),
	DayFromYear: require('es-abstract/2017/DayFromYear'),
	DaysInYear: require('es-abstract/2017/DaysInYear'),
	DayWithinYear: require('es-abstract/2017/DayWithinYear'),
	DefinePropertyOrThrow: require('es-abstract/2017/DefinePropertyOrThrow'),
	DeletePropertyOrThrow: require('es-abstract/2017/DeletePropertyOrThrow'),
	DetachArrayBuffer: require('es-abstract/2017/DetachArrayBuffer'),
	EnumerableOwnProperties: require('es-abstract/2017/EnumerableOwnProperties'),
	floor: require('es-abstract/2017/floor'),
	FromPropertyDescriptor: require('es-abstract/2017/FromPropertyDescriptor'),
	Get: require('es-abstract/2017/Get'),
	GetGlobalObject: require('es-abstract/2017/GetGlobalObject'),
	GetIterator: require('es-abstract/2017/GetIterator'),
	GetMethod: require('es-abstract/2017/GetMethod'),
	GetOwnPropertyKeys: require('es-abstract/2017/GetOwnPropertyKeys'),
	GetPrototypeFromConstructor: require('es-abstract/2017/GetPrototypeFromConstructor'),
	GetSubstitution: require('es-abstract/2017/GetSubstitution'),
	GetV: require('es-abstract/2017/GetV'),
	GetValueFromBuffer: require('es-abstract/2017/GetValueFromBuffer'),
	HasOwnProperty: require('es-abstract/2017/HasOwnProperty'),
	HasProperty: require('es-abstract/2017/HasProperty'),
	HourFromTime: require('es-abstract/2017/HourFromTime'),
	InLeapYear: require('es-abstract/2017/InLeapYear'),
	InstanceofOperator: require('es-abstract/2017/InstanceofOperator'),
	Invoke: require('es-abstract/2017/Invoke'),
	IsAccessorDescriptor: require('es-abstract/2017/IsAccessorDescriptor'),
	IsArray: require('es-abstract/2017/IsArray'),
	IsCallable: require('es-abstract/2017/IsCallable'),
	IsCompatiblePropertyDescriptor: require('es-abstract/2017/IsCompatiblePropertyDescriptor'),
	IsConcatSpreadable: require('es-abstract/2017/IsConcatSpreadable'),
	IsConstructor: require('es-abstract/2017/IsConstructor'),
	IsDataDescriptor: require('es-abstract/2017/IsDataDescriptor'),
	IsDetachedBuffer: require('es-abstract/2017/IsDetachedBuffer'),
	IsExtensible: require('es-abstract/2017/IsExtensible'),
	IsGenericDescriptor: require('es-abstract/2017/IsGenericDescriptor'),
	IsInteger: require('es-abstract/2017/IsInteger'),
	IsPromise: require('es-abstract/2017/IsPromise'),
	IsPropertyDescriptor: require('es-abstract/2017/IsPropertyDescriptor'),
	IsPropertyKey: require('es-abstract/2017/IsPropertyKey'),
	IsRegExp: require('es-abstract/2017/IsRegExp'),
	IsSharedArrayBuffer: require('es-abstract/2017/IsSharedArrayBuffer'),
	IsWordChar: require('es-abstract/2017/IsWordChar'),
	IterableToList: require('es-abstract/2017/IterableToList'),
	IteratorClose: require('es-abstract/2017/IteratorClose'),
	IteratorComplete: require('es-abstract/2017/IteratorComplete'),
	IteratorNext: require('es-abstract/2017/IteratorNext'),
	IteratorStep: require('es-abstract/2017/IteratorStep'),
	IteratorValue: require('es-abstract/2017/IteratorValue'),
	MakeDate: require('es-abstract/2017/MakeDate'),
	MakeDay: require('es-abstract/2017/MakeDay'),
	MakeTime: require('es-abstract/2017/MakeTime'),
	max: require('es-abstract/2017/max'),
	min: require('es-abstract/2017/min'),
	MinFromTime: require('es-abstract/2017/MinFromTime'),
	modulo: require('es-abstract/2017/modulo'),
	MonthFromTime: require('es-abstract/2017/MonthFromTime'),
	msFromTime: require('es-abstract/2017/msFromTime'),
	NewPromiseCapability: require('es-abstract/2017/NewPromiseCapability'),
	NormalCompletion: require('es-abstract/2017/NormalCompletion'),
	NumberToRawBytes: require('es-abstract/2017/NumberToRawBytes'),
	ObjectCreate: require('es-abstract/2017/ObjectCreate'),
	ObjectDefineProperties: require('es-abstract/2017/ObjectDefineProperties'),
	OrdinaryCreateFromConstructor: require('es-abstract/2017/OrdinaryCreateFromConstructor'),
	OrdinaryDefineOwnProperty: require('es-abstract/2017/OrdinaryDefineOwnProperty'),
	OrdinaryGetOwnProperty: require('es-abstract/2017/OrdinaryGetOwnProperty'),
	OrdinaryGetPrototypeOf: require('es-abstract/2017/OrdinaryGetPrototypeOf'),
	OrdinaryHasInstance: require('es-abstract/2017/OrdinaryHasInstance'),
	OrdinaryHasProperty: require('es-abstract/2017/OrdinaryHasProperty'),
	OrdinarySetPrototypeOf: require('es-abstract/2017/OrdinarySetPrototypeOf'),
	OrdinaryToPrimitive: require('es-abstract/2017/OrdinaryToPrimitive'),
	QuoteJSONString: require('es-abstract/2017/QuoteJSONString'),
	RawBytesToNumber: require('es-abstract/2017/RawBytesToNumber'),
	RegExpCreate: require('es-abstract/2017/RegExpCreate'),
	RegExpExec: require('es-abstract/2017/RegExpExec'),
	RequireObjectCoercible: require('es-abstract/2017/RequireObjectCoercible'),
	SameValue: require('es-abstract/2017/SameValue'),
	SameValueNonNumber: require('es-abstract/2017/SameValueNonNumber'),
	SameValueZero: require('es-abstract/2017/SameValueZero'),
	SecFromTime: require('es-abstract/2017/SecFromTime'),
	Set: require('es-abstract/2017/Set'),
	SetFunctionName: require('es-abstract/2017/SetFunctionName'),
	SetIntegrityLevel: require('es-abstract/2017/SetIntegrityLevel'),
	SetValueInBuffer: require('es-abstract/2017/SetValueInBuffer'),
	SpeciesConstructor: require('es-abstract/2017/SpeciesConstructor'),
	SplitMatch: require('es-abstract/2017/SplitMatch'),
	StringCreate: require('es-abstract/2017/StringCreate'),
	StringGetOwnProperty: require('es-abstract/2017/StringGetOwnProperty'),
	SymbolDescriptiveString: require('es-abstract/2017/SymbolDescriptiveString'),
	TestIntegrityLevel: require('es-abstract/2017/TestIntegrityLevel'),
	thisBooleanValue: require('es-abstract/2017/thisBooleanValue'),
	thisNumberValue: require('es-abstract/2017/thisNumberValue'),
	thisStringValue: require('es-abstract/2017/thisStringValue'),
	thisTimeValue: require('es-abstract/2017/thisTimeValue'),
	TimeClip: require('es-abstract/2017/TimeClip'),
	TimeFromYear: require('es-abstract/2017/TimeFromYear'),
	TimeWithinDay: require('es-abstract/2017/TimeWithinDay'),
	ToBoolean: require('es-abstract/2017/ToBoolean'),
	ToDateString: require('es-abstract/2017/ToDateString'),
	ToIndex: require('es-abstract/2017/ToIndex'),
	ToInt16: require('es-abstract/2017/ToInt16'),
	ToInt32: require('es-abstract/2017/ToInt32'),
	ToInt8: require('es-abstract/2017/ToInt8'),
	ToInteger: require('es-abstract/2017/ToInteger'),
	ToLength: require('es-abstract/2017/ToLength'),
	ToNumber: require('es-abstract/2017/ToNumber'),
	ToObject: require('es-abstract/2017/ToObject'),
	ToPrimitive: require('es-abstract/2017/ToPrimitive'),
	ToPropertyDescriptor: require('es-abstract/2017/ToPropertyDescriptor'),
	ToPropertyKey: require('es-abstract/2017/ToPropertyKey'),
	ToString: require('es-abstract/2017/ToString'),
	ToUint16: require('es-abstract/2017/ToUint16'),
	ToUint32: require('es-abstract/2017/ToUint32'),
	ToUint8: require('es-abstract/2017/ToUint8'),
	ToUint8Clamp: require('es-abstract/2017/ToUint8Clamp'),
	Type: require('es-abstract/2017/Type'),
	TypedArrayCreate: require('es-abstract/2017/TypedArrayCreate'),
	TypedArraySpeciesCreate: require('es-abstract/2017/TypedArraySpeciesCreate'),
	UTF16Decode: require('es-abstract/2017/UTF16Decode'),
	UTF16Encoding: require('es-abstract/2017/UTF16Encoding'),
	ValidateAndApplyPropertyDescriptor: require('es-abstract/2017/ValidateAndApplyPropertyDescriptor'),
	ValidateAtomicAccess: require('es-abstract/2017/ValidateAtomicAccess'),
	ValidateTypedArray: require('es-abstract/2017/ValidateTypedArray'),
	WeekDay: require('es-abstract/2017/WeekDay'),
	WordCharacters: require('es-abstract/2017/WordCharacters'),
	YearFromTime: require('es-abstract/2017/YearFromTime')
};

module.exports = ES2017;
