// es-abstract/es2023.js
'use strict';

/* eslint global-require: 0 */
// https://262.ecma-international.org/14.0/#sec-abstract-operations
var ES2023 = {
	abs: require('es-abstract/2023/abs'),
	AddEntriesFromIterable: require('es-abstract/2023/AddEntriesFromIterable'),
	AddToKeptObjects: require('es-abstract/2023/AddToKeptObjects'),
	AdvanceStringIndex: require('es-abstract/2023/AdvanceStringIndex'),
	ApplyStringOrNumericBinaryOperator: require('es-abstract/2023/ApplyStringOrNumericBinaryOperator'),
	ArrayCreate: require('es-abstract/2023/ArrayCreate'),
	ArraySetLength: require('es-abstract/2023/ArraySetLength'),
	ArraySpeciesCreate: require('es-abstract/2023/ArraySpeciesCreate'),
	AsyncFromSyncIteratorContinuation: require('es-abstract/2023/AsyncFromSyncIteratorContinuation'),
	AsyncIteratorClose: require('es-abstract/2023/AsyncIteratorClose'),
	BigInt: require('es-abstract/2023/BigInt'),
	BigIntBitwiseOp: require('es-abstract/2023/BigIntBitwiseOp'),
	BinaryAnd: require('es-abstract/2023/BinaryAnd'),
	BinaryOr: require('es-abstract/2023/BinaryOr'),
	BinaryXor: require('es-abstract/2023/BinaryXor'),
	ByteListBitwiseOp: require('es-abstract/2023/ByteListBitwiseOp'),
	ByteListEqual: require('es-abstract/2023/ByteListEqual'),
	Call: require('es-abstract/2023/Call'),
	CanBeHeldWeakly: require('es-abstract/2023/CanBeHeldWeakly'),
	Canonicalize: require('es-abstract/2023/Canonicalize'),
	CanonicalNumericIndexString: require('es-abstract/2023/CanonicalNumericIndexString'),
	CharacterRange: require('es-abstract/2023/CharacterRange'),
	clamp: require('es-abstract/2023/clamp'),
	ClearKeptObjects: require('es-abstract/2023/ClearKeptObjects'),
	CloneArrayBuffer: require('es-abstract/2023/CloneArrayBuffer'),
	CodePointAt: require('es-abstract/2023/CodePointAt'),
	CodePointsToString: require('es-abstract/2023/CodePointsToString'),
	CompareArrayElements: require('es-abstract/2023/CompareArrayElements'),
	CompareTypedArrayElements: require('es-abstract/2023/CompareTypedArrayElements'),
	CompletePropertyDescriptor: require('es-abstract/2023/CompletePropertyDescriptor'),
	CompletionRecord: require('es-abstract/2023/CompletionRecord'),
	CopyDataProperties: require('es-abstract/2023/CopyDataProperties'),
	CreateAsyncFromSyncIterator: require('es-abstract/2023/CreateAsyncFromSyncIterator'),
	CreateDataProperty: require('es-abstract/2023/CreateDataProperty'),
	CreateDataPropertyOrThrow: require('es-abstract/2023/CreateDataPropertyOrThrow'),
	CreateHTML: require('es-abstract/2023/CreateHTML'),
	CreateIterResultObject: require('es-abstract/2023/CreateIterResultObject'),
	CreateListFromArrayLike: require('es-abstract/2023/CreateListFromArrayLike'),
	CreateMethodProperty: require('es-abstract/2023/CreateMethodProperty'),
	CreateNonEnumerableDataPropertyOrThrow: require('es-abstract/2023/CreateNonEnumerableDataPropertyOrThrow'),
	CreateRegExpStringIterator: require('es-abstract/2023/CreateRegExpStringIterator'),
	DateFromTime: require('es-abstract/2023/DateFromTime'),
	DateString: require('es-abstract/2023/DateString'),
	Day: require('es-abstract/2023/Day'),
	DayFromYear: require('es-abstract/2023/DayFromYear'),
	DaysInYear: require('es-abstract/2023/DaysInYear'),
	DayWithinYear: require('es-abstract/2023/DayWithinYear'),
	DefinePropertyOrThrow: require('es-abstract/2023/DefinePropertyOrThrow'),
	DeletePropertyOrThrow: require('es-abstract/2023/DeletePropertyOrThrow'),
	DetachArrayBuffer: require('es-abstract/2023/DetachArrayBuffer'),
	EnumerableOwnProperties: require('es-abstract/2023/EnumerableOwnProperties'),
	FindViaPredicate: require('es-abstract/2023/FindViaPredicate'),
	FlattenIntoArray: require('es-abstract/2023/FlattenIntoArray'),
	floor: require('es-abstract/2023/floor'),
	FromPropertyDescriptor: require('es-abstract/2023/FromPropertyDescriptor'),
	Get: require('es-abstract/2023/Get'),
	GetGlobalObject: require('es-abstract/2023/GetGlobalObject'),
	GetIterator: require('es-abstract/2023/GetIterator'),
	GetIteratorFromMethod: require('es-abstract/2023/GetIteratorFromMethod'),
	GetMatchIndexPair: require('es-abstract/2023/GetMatchIndexPair'),
	GetMatchString: require('es-abstract/2023/GetMatchString'),
	GetMethod: require('es-abstract/2023/GetMethod'),
	GetOwnPropertyKeys: require('es-abstract/2023/GetOwnPropertyKeys'),
	GetPromiseResolve: require('es-abstract/2023/GetPromiseResolve'),
	GetPrototypeFromConstructor: require('es-abstract/2023/GetPrototypeFromConstructor'),
	GetStringIndex: require('es-abstract/2023/GetStringIndex'),
	GetSubstitution: require('es-abstract/2023/GetSubstitution'),
	GetV: require('es-abstract/2023/GetV'),
	GetValueFromBuffer: require('es-abstract/2023/GetValueFromBuffer'),
	HasOwnProperty: require('es-abstract/2023/HasOwnProperty'),
	HasProperty: require('es-abstract/2023/HasProperty'),
	HourFromTime: require('es-abstract/2023/HourFromTime'),
	InLeapYear: require('es-abstract/2023/InLeapYear'),
	InstallErrorCause: require('es-abstract/2023/InstallErrorCause'),
	InstanceofOperator: require('es-abstract/2023/InstanceofOperator'),
	Invoke: require('es-abstract/2023/Invoke'),
	IsAccessorDescriptor: require('es-abstract/2023/IsAccessorDescriptor'),
	IsArray: require('es-abstract/2023/IsArray'),
	IsBigIntElementType: require('es-abstract/2023/IsBigIntElementType'),
	IsCallable: require('es-abstract/2023/IsCallable'),
	IsCompatiblePropertyDescriptor: require('es-abstract/2023/IsCompatiblePropertyDescriptor'),
	IsConcatSpreadable: require('es-abstract/2023/IsConcatSpreadable'),
	IsConstructor: require('es-abstract/2023/IsConstructor'),
	IsDataDescriptor: require('es-abstract/2023/IsDataDescriptor'),
	IsDetachedBuffer: require('es-abstract/2023/IsDetachedBuffer'),
	IsExtensible: require('es-abstract/2023/IsExtensible'),
	IsGenericDescriptor: require('es-abstract/2023/IsGenericDescriptor'),
	IsIntegralNumber: require('es-abstract/2023/IsIntegralNumber'),
	IsLessThan: require('es-abstract/2023/IsLessThan'),
	IsLooselyEqual: require('es-abstract/2023/IsLooselyEqual'),
	IsNoTearConfiguration: require('es-abstract/2023/IsNoTearConfiguration'),
	IsPromise: require('es-abstract/2023/IsPromise'),
	IsPropertyKey: require('es-abstract/2023/IsPropertyKey'),
	IsRegExp: require('es-abstract/2023/IsRegExp'),
	IsSharedArrayBuffer: require('es-abstract/2023/IsSharedArrayBuffer'),
	IsStrictlyEqual: require('es-abstract/2023/IsStrictlyEqual'),
	IsStringWellFormedUnicode: require('es-abstract/2023/IsStringWellFormedUnicode'),
	IsUnclampedIntegerElementType: require('es-abstract/2023/IsUnclampedIntegerElementType'),
	IsUnsignedElementType: require('es-abstract/2023/IsUnsignedElementType'),
	IsValidIntegerIndex: require('es-abstract/2023/IsValidIntegerIndex'),
	IsWordChar: require('es-abstract/2023/IsWordChar'),
	IteratorClose: require('es-abstract/2023/IteratorClose'),
	IteratorComplete: require('es-abstract/2023/IteratorComplete'),
	IteratorNext: require('es-abstract/2023/IteratorNext'),
	IteratorStep: require('es-abstract/2023/IteratorStep'),
	IteratorToList: require('es-abstract/2023/IteratorToList'),
	IteratorValue: require('es-abstract/2023/IteratorValue'),
	KeyForSymbol: require('es-abstract/2023/KeyForSymbol'),
	LengthOfArrayLike: require('es-abstract/2023/LengthOfArrayLike'),
	MakeDate: require('es-abstract/2023/MakeDate'),
	MakeDay: require('es-abstract/2023/MakeDay'),
	MakeMatchIndicesIndexPairArray: require('es-abstract/2023/MakeMatchIndicesIndexPairArray'),
	MakeTime: require('es-abstract/2023/MakeTime'),
	max: require('es-abstract/2023/max'),
	min: require('es-abstract/2023/min'),
	MinFromTime: require('es-abstract/2023/MinFromTime'),
	modulo: require('es-abstract/2023/modulo'),
	MonthFromTime: require('es-abstract/2023/MonthFromTime'),
	msFromTime: require('es-abstract/2023/msFromTime'),
	NewPromiseCapability: require('es-abstract/2023/NewPromiseCapability'),
	NormalCompletion: require('es-abstract/2023/NormalCompletion'),
	Number: require('es-abstract/2023/Number'),
	NumberBitwiseOp: require('es-abstract/2023/NumberBitwiseOp'),
	NumberToBigInt: require('es-abstract/2023/NumberToBigInt'),
	NumericToRawBytes: require('es-abstract/2023/NumericToRawBytes'),
	ObjectDefineProperties: require('es-abstract/2023/ObjectDefineProperties'),
	OrdinaryCreateFromConstructor: require('es-abstract/2023/OrdinaryCreateFromConstructor'),
	OrdinaryDefineOwnProperty: require('es-abstract/2023/OrdinaryDefineOwnProperty'),
	OrdinaryGetOwnProperty: require('es-abstract/2023/OrdinaryGetOwnProperty'),
	OrdinaryGetPrototypeOf: require('es-abstract/2023/OrdinaryGetPrototypeOf'),
	OrdinaryHasInstance: require('es-abstract/2023/OrdinaryHasInstance'),
	OrdinaryHasProperty: require('es-abstract/2023/OrdinaryHasProperty'),
	OrdinaryObjectCreate: require('es-abstract/2023/OrdinaryObjectCreate'),
	OrdinarySetPrototypeOf: require('es-abstract/2023/OrdinarySetPrototypeOf'),
	OrdinaryToPrimitive: require('es-abstract/2023/OrdinaryToPrimitive'),
	ParseHexOctet: require('es-abstract/2023/ParseHexOctet'),
	PromiseResolve: require('es-abstract/2023/PromiseResolve'),
	QuoteJSONString: require('es-abstract/2023/QuoteJSONString'),
	RawBytesToNumeric: require('es-abstract/2023/RawBytesToNumeric'),
	RegExpCreate: require('es-abstract/2023/RegExpCreate'),
	RegExpExec: require('es-abstract/2023/RegExpExec'),
	RegExpHasFlag: require('es-abstract/2023/RegExpHasFlag'),
	RequireObjectCoercible: require('es-abstract/2023/RequireObjectCoercible'),
	SameValue: require('es-abstract/2023/SameValue'),
	SameValueNonNumber: require('es-abstract/2023/SameValueNonNumber'),
	SameValueZero: require('es-abstract/2023/SameValueZero'),
	SecFromTime: require('es-abstract/2023/SecFromTime'),
	Set: require('es-abstract/2023/Set'),
	SetFunctionLength: require('es-abstract/2023/SetFunctionLength'),
	SetFunctionName: require('es-abstract/2023/SetFunctionName'),
	SetIntegrityLevel: require('es-abstract/2023/SetIntegrityLevel'),
	SetTypedArrayFromArrayLike: require('es-abstract/2023/SetTypedArrayFromArrayLike'),
	SetTypedArrayFromTypedArray: require('es-abstract/2023/SetTypedArrayFromTypedArray'),
	SetValueInBuffer: require('es-abstract/2023/SetValueInBuffer'),
	SortIndexedProperties: require('es-abstract/2023/SortIndexedProperties'),
	SpeciesConstructor: require('es-abstract/2023/SpeciesConstructor'),
	StringCreate: require('es-abstract/2023/StringCreate'),
	StringGetOwnProperty: require('es-abstract/2023/StringGetOwnProperty'),
	StringIndexOf: require('es-abstract/2023/StringIndexOf'),
	StringPad: require('es-abstract/2023/StringPad'),
	StringToBigInt: require('es-abstract/2023/StringToBigInt'),
	StringToCodePoints: require('es-abstract/2023/StringToCodePoints'),
	StringToNumber: require('es-abstract/2023/StringToNumber'),
	substring: require('es-abstract/2023/substring'),
	SymbolDescriptiveString: require('es-abstract/2023/SymbolDescriptiveString'),
	TestIntegrityLevel: require('es-abstract/2023/TestIntegrityLevel'),
	thisBigIntValue: require('es-abstract/2023/thisBigIntValue'),
	thisBooleanValue: require('es-abstract/2023/thisBooleanValue'),
	thisNumberValue: require('es-abstract/2023/thisNumberValue'),
	thisStringValue: require('es-abstract/2023/thisStringValue'),
	thisSymbolValue: require('es-abstract/2023/thisSymbolValue'),
	thisTimeValue: require('es-abstract/2023/thisTimeValue'),
	ThrowCompletion: require('es-abstract/2023/ThrowCompletion'),
	TimeClip: require('es-abstract/2023/TimeClip'),
	TimeFromYear: require('es-abstract/2023/TimeFromYear'),
	TimeString: require('es-abstract/2023/TimeString'),
	TimeWithinDay: require('es-abstract/2023/TimeWithinDay'),
	ToBigInt: require('es-abstract/2023/ToBigInt'),
	ToBigInt64: require('es-abstract/2023/ToBigInt64'),
	ToBigUint64: require('es-abstract/2023/ToBigUint64'),
	ToBoolean: require('es-abstract/2023/ToBoolean'),
	ToDateString: require('es-abstract/2023/ToDateString'),
	ToIndex: require('es-abstract/2023/ToIndex'),
	ToInt16: require('es-abstract/2023/ToInt16'),
	ToInt32: require('es-abstract/2023/ToInt32'),
	ToInt8: require('es-abstract/2023/ToInt8'),
	ToIntegerOrInfinity: require('es-abstract/2023/ToIntegerOrInfinity'),
	ToLength: require('es-abstract/2023/ToLength'),
	ToNumber: require('es-abstract/2023/ToNumber'),
	ToNumeric: require('es-abstract/2023/ToNumeric'),
	ToObject: require('es-abstract/2023/ToObject'),
	ToPrimitive: require('es-abstract/2023/ToPrimitive'),
	ToPropertyDescriptor: require('es-abstract/2023/ToPropertyDescriptor'),
	ToPropertyKey: require('es-abstract/2023/ToPropertyKey'),
	ToString: require('es-abstract/2023/ToString'),
	ToUint16: require('es-abstract/2023/ToUint16'),
	ToUint32: require('es-abstract/2023/ToUint32'),
	ToUint8: require('es-abstract/2023/ToUint8'),
	ToUint8Clamp: require('es-abstract/2023/ToUint8Clamp'),
	ToZeroPaddedDecimalString: require('es-abstract/2023/ToZeroPaddedDecimalString'),
	TrimString: require('es-abstract/2023/TrimString'),
	truncate: require('es-abstract/2023/truncate'),
	Type: require('es-abstract/2023/Type'),
	TypedArrayCreate: require('es-abstract/2023/TypedArrayCreate'),
	TypedArrayCreateSameType: require('es-abstract/2023/TypedArrayCreateSameType'),
	TypedArrayElementSize: require('es-abstract/2023/TypedArrayElementSize'),
	TypedArrayElementType: require('es-abstract/2023/TypedArrayElementType'),
	TypedArraySpeciesCreate: require('es-abstract/2023/TypedArraySpeciesCreate'),
	UnicodeEscape: require('es-abstract/2023/UnicodeEscape'),
	UTF16EncodeCodePoint: require('es-abstract/2023/UTF16EncodeCodePoint'),
	UTF16SurrogatePairToCodePoint: require('es-abstract/2023/UTF16SurrogatePairToCodePoint'),
	ValidateAndApplyPropertyDescriptor: require('es-abstract/2023/ValidateAndApplyPropertyDescriptor'),
	ValidateAtomicAccess: require('es-abstract/2023/ValidateAtomicAccess'),
	ValidateIntegerTypedArray: require('es-abstract/2023/ValidateIntegerTypedArray'),
	ValidateTypedArray: require('es-abstract/2023/ValidateTypedArray'),
	WeakRefDeref: require('es-abstract/2023/WeakRefDeref'),
	WeekDay: require('es-abstract/2023/WeekDay'),
	WordCharacters: require('es-abstract/2023/WordCharacters'),
	YearFromTime: require('es-abstract/2023/YearFromTime')
};

module.exports = ES2023;
