// es-abstract/es2018.js
'use strict';

/* eslint global-require: 0 */
// https://262.ecma-international.org/9.0/#sec-abstract-operations
var ES2018 = {
	'Abstract Equality Comparison': require('es-abstract/2018/AbstractEqualityComparison'),
	'Abstract Relational Comparison': require('es-abstract/2018/AbstractRelationalComparison'),
	'Strict Equality Comparison': require('es-abstract/2018/StrictEqualityComparison'),
	abs: require('es-abstract/2018/abs'),
	AdvanceStringIndex: require('es-abstract/2018/AdvanceStringIndex'),
	ArrayCreate: require('es-abstract/2018/ArrayCreate'),
	ArraySetLength: require('es-abstract/2018/ArraySetLength'),
	ArraySpeciesCreate: require('es-abstract/2018/ArraySpeciesCreate'),
	AsyncIteratorClose: require('es-abstract/2018/AsyncIteratorClose'),
	Call: require('es-abstract/2018/Call'),
	Canonicalize: require('es-abstract/2018/Canonicalize'),
	CanonicalNumericIndexString: require('es-abstract/2018/CanonicalNumericIndexString'),
	CharacterRange: require('es-abstract/2018/CharacterRange'),
	CompletePropertyDescriptor: require('es-abstract/2018/CompletePropertyDescriptor'),
	CompletionRecord: require('es-abstract/2018/CompletionRecord'),
	CopyDataProperties: require('es-abstract/2018/CopyDataProperties'),
	CreateAsyncFromSyncIterator: require('es-abstract/2018/CreateAsyncFromSyncIterator'),
	CreateDataProperty: require('es-abstract/2018/CreateDataProperty'),
	CreateDataPropertyOrThrow: require('es-abstract/2018/CreateDataPropertyOrThrow'),
	CreateHTML: require('es-abstract/2018/CreateHTML'),
	CreateIterResultObject: require('es-abstract/2018/CreateIterResultObject'),
	CreateListFromArrayLike: require('es-abstract/2018/CreateListFromArrayLike'),
	CreateMethodProperty: require('es-abstract/2018/CreateMethodProperty'),
	DateFromTime: require('es-abstract/2018/DateFromTime'),
	DateString: require('es-abstract/2018/DateString'),
	Day: require('es-abstract/2018/Day'),
	DayFromYear: require('es-abstract/2018/DayFromYear'),
	DaysInYear: require('es-abstract/2018/DaysInYear'),
	DayWithinYear: require('es-abstract/2018/DayWithinYear'),
	DefinePropertyOrThrow: require('es-abstract/2018/DefinePropertyOrThrow'),
	DeletePropertyOrThrow: require('es-abstract/2018/DeletePropertyOrThrow'),
	DetachArrayBuffer: require('es-abstract/2018/DetachArrayBuffer'),
	EnumerableOwnPropertyNames: require('es-abstract/2018/EnumerableOwnPropertyNames'),
	floor: require('es-abstract/2018/floor'),
	FromPropertyDescriptor: require('es-abstract/2018/FromPropertyDescriptor'),
	Get: require('es-abstract/2018/Get'),
	GetGlobalObject: require('es-abstract/2018/GetGlobalObject'),
	GetIterator: require('es-abstract/2018/GetIterator'),
	GetMethod: require('es-abstract/2018/GetMethod'),
	GetOwnPropertyKeys: require('es-abstract/2018/GetOwnPropertyKeys'),
	GetPrototypeFromConstructor: require('es-abstract/2018/GetPrototypeFromConstructor'),
	GetSubstitution: require('es-abstract/2018/GetSubstitution'),
	GetV: require('es-abstract/2018/GetV'),
	GetValueFromBuffer: require('es-abstract/2018/GetValueFromBuffer'),
	HasOwnProperty: require('es-abstract/2018/HasOwnProperty'),
	HasProperty: require('es-abstract/2018/HasProperty'),
	HourFromTime: require('es-abstract/2018/HourFromTime'),
	InLeapYear: require('es-abstract/2018/InLeapYear'),
	InstanceofOperator: require('es-abstract/2018/InstanceofOperator'),
	Invoke: require('es-abstract/2018/Invoke'),
	IsAccessorDescriptor: require('es-abstract/2018/IsAccessorDescriptor'),
	IsArray: require('es-abstract/2018/IsArray'),
	IsCallable: require('es-abstract/2018/IsCallable'),
	IsCompatiblePropertyDescriptor: require('es-abstract/2018/IsCompatiblePropertyDescriptor'),
	IsConcatSpreadable: require('es-abstract/2018/IsConcatSpreadable'),
	IsConstructor: require('es-abstract/2018/IsConstructor'),
	IsDataDescriptor: require('es-abstract/2018/IsDataDescriptor'),
	IsDetachedBuffer: require('es-abstract/2018/IsDetachedBuffer'),
	IsExtensible: require('es-abstract/2018/IsExtensible'),
	IsGenericDescriptor: require('es-abstract/2018/IsGenericDescriptor'),
	IsInteger: require('es-abstract/2018/IsInteger'),
	IsPromise: require('es-abstract/2018/IsPromise'),
	IsPropertyKey: require('es-abstract/2018/IsPropertyKey'),
	IsRegExp: require('es-abstract/2018/IsRegExp'),
	IsSharedArrayBuffer: require('es-abstract/2018/IsSharedArrayBuffer'),
	IsStringPrefix: require('es-abstract/2018/IsStringPrefix'),
	IsWordChar: require('es-abstract/2018/IsWordChar'),
	IterableToList: require('es-abstract/2018/IterableToList'),
	IteratorClose: require('es-abstract/2018/IteratorClose'),
	IteratorComplete: require('es-abstract/2018/IteratorComplete'),
	IteratorNext: require('es-abstract/2018/IteratorNext'),
	IteratorStep: require('es-abstract/2018/IteratorStep'),
	IteratorValue: require('es-abstract/2018/IteratorValue'),
	MakeDate: require('es-abstract/2018/MakeDate'),
	MakeDay: require('es-abstract/2018/MakeDay'),
	MakeTime: require('es-abstract/2018/MakeTime'),
	max: require('es-abstract/2018/max'),
	min: require('es-abstract/2018/min'),
	MinFromTime: require('es-abstract/2018/MinFromTime'),
	modulo: require('es-abstract/2018/modulo'),
	MonthFromTime: require('es-abstract/2018/MonthFromTime'),
	msFromTime: require('es-abstract/2018/msFromTime'),
	NewPromiseCapability: require('es-abstract/2018/NewPromiseCapability'),
	NormalCompletion: require('es-abstract/2018/NormalCompletion'),
	NumberToRawBytes: require('es-abstract/2018/NumberToRawBytes'),
	NumberToString: require('es-abstract/2018/NumberToString'),
	ObjectCreate: require('es-abstract/2018/ObjectCreate'),
	ObjectDefineProperties: require('es-abstract/2018/ObjectDefineProperties'),
	OrdinaryCreateFromConstructor: require('es-abstract/2018/OrdinaryCreateFromConstructor'),
	OrdinaryDefineOwnProperty: require('es-abstract/2018/OrdinaryDefineOwnProperty'),
	OrdinaryGetOwnProperty: require('es-abstract/2018/OrdinaryGetOwnProperty'),
	OrdinaryGetPrototypeOf: require('es-abstract/2018/OrdinaryGetPrototypeOf'),
	OrdinaryHasInstance: require('es-abstract/2018/OrdinaryHasInstance'),
	OrdinaryHasProperty: require('es-abstract/2018/OrdinaryHasProperty'),
	OrdinarySetPrototypeOf: require('es-abstract/2018/OrdinarySetPrototypeOf'),
	OrdinaryToPrimitive: require('es-abstract/2018/OrdinaryToPrimitive'),
	PromiseResolve: require('es-abstract/2018/PromiseResolve'),
	QuoteJSONString: require('es-abstract/2018/QuoteJSONString'),
	RawBytesToNumber: require('es-abstract/2018/RawBytesToNumber'),
	RegExpCreate: require('es-abstract/2018/RegExpCreate'),
	RegExpExec: require('es-abstract/2018/RegExpExec'),
	RequireObjectCoercible: require('es-abstract/2018/RequireObjectCoercible'),
	SameValue: require('es-abstract/2018/SameValue'),
	SameValueNonNumber: require('es-abstract/2018/SameValueNonNumber'),
	SameValueZero: require('es-abstract/2018/SameValueZero'),
	SecFromTime: require('es-abstract/2018/SecFromTime'),
	Set: require('es-abstract/2018/Set'),
	SetFunctionLength: require('es-abstract/2018/SetFunctionLength'),
	SetFunctionName: require('es-abstract/2018/SetFunctionName'),
	SetIntegrityLevel: require('es-abstract/2018/SetIntegrityLevel'),
	SetValueInBuffer: require('es-abstract/2018/SetValueInBuffer'),
	SpeciesConstructor: require('es-abstract/2018/SpeciesConstructor'),
	SplitMatch: require('es-abstract/2018/SplitMatch'),
	StringCreate: require('es-abstract/2018/StringCreate'),
	StringGetOwnProperty: require('es-abstract/2018/StringGetOwnProperty'),
	SymbolDescriptiveString: require('es-abstract/2018/SymbolDescriptiveString'),
	TestIntegrityLevel: require('es-abstract/2018/TestIntegrityLevel'),
	thisBooleanValue: require('es-abstract/2018/thisBooleanValue'),
	thisNumberValue: require('es-abstract/2018/thisNumberValue'),
	thisStringValue: require('es-abstract/2018/thisStringValue'),
	thisSymbolValue: require('es-abstract/2018/thisSymbolValue'),
	thisTimeValue: require('es-abstract/2018/thisTimeValue'),
	ThrowCompletion: require('es-abstract/2018/ThrowCompletion'),
	TimeClip: require('es-abstract/2018/TimeClip'),
	TimeFromYear: require('es-abstract/2018/TimeFromYear'),
	TimeString: require('es-abstract/2018/TimeString'),
	TimeWithinDay: require('es-abstract/2018/TimeWithinDay'),
	ToBoolean: require('es-abstract/2018/ToBoolean'),
	ToDateString: require('es-abstract/2018/ToDateString'),
	ToIndex: require('es-abstract/2018/ToIndex'),
	ToInt16: require('es-abstract/2018/ToInt16'),
	ToInt32: require('es-abstract/2018/ToInt32'),
	ToInt8: require('es-abstract/2018/ToInt8'),
	ToInteger: require('es-abstract/2018/ToInteger'),
	ToLength: require('es-abstract/2018/ToLength'),
	ToNumber: require('es-abstract/2018/ToNumber'),
	ToObject: require('es-abstract/2018/ToObject'),
	ToPrimitive: require('es-abstract/2018/ToPrimitive'),
	ToPropertyDescriptor: require('es-abstract/2018/ToPropertyDescriptor'),
	ToPropertyKey: require('es-abstract/2018/ToPropertyKey'),
	ToString: require('es-abstract/2018/ToString'),
	ToUint16: require('es-abstract/2018/ToUint16'),
	ToUint32: require('es-abstract/2018/ToUint32'),
	ToUint8: require('es-abstract/2018/ToUint8'),
	ToUint8Clamp: require('es-abstract/2018/ToUint8Clamp'),
	Type: require('es-abstract/2018/Type'),
	TypedArrayCreate: require('es-abstract/2018/TypedArrayCreate'),
	TypedArraySpeciesCreate: require('es-abstract/2018/TypedArraySpeciesCreate'),
	UnicodeEscape: require('es-abstract/2018/UnicodeEscape'),
	UTF16Decode: require('es-abstract/2018/UTF16Decode'),
	UTF16Encoding: require('es-abstract/2018/UTF16Encoding'),
	ValidateAndApplyPropertyDescriptor: require('es-abstract/2018/ValidateAndApplyPropertyDescriptor'),
	ValidateAtomicAccess: require('es-abstract/2018/ValidateAtomicAccess'),
	ValidateTypedArray: require('es-abstract/2018/ValidateTypedArray'),
	WeekDay: require('es-abstract/2018/WeekDay'),
	WordCharacters: require('es-abstract/2018/WordCharacters'),
	YearFromTime: require('es-abstract/2018/YearFromTime')
};

module.exports = ES2018;
