// es-abstract/es2022.js
'use strict';

/* eslint global-require: 0 */
// https://262.ecma-international.org/13.0/#sec-abstract-operations
var ES2022 = {
	abs: require('es-abstract/2022/abs'),
	AddEntriesFromIterable: require('es-abstract/2022/AddEntriesFromIterable'),
	AddToKeptObjects: require('es-abstract/2022/AddToKeptObjects'),
	AdvanceStringIndex: require('es-abstract/2022/AdvanceStringIndex'),
	ApplyStringOrNumericBinaryOperator: require('es-abstract/2022/ApplyStringOrNumericBinaryOperator'),
	ArrayCreate: require('es-abstract/2022/ArrayCreate'),
	ArraySetLength: require('es-abstract/2022/ArraySetLength'),
	ArraySpeciesCreate: require('es-abstract/2022/ArraySpeciesCreate'),
	AsyncFromSyncIteratorContinuation: require('es-abstract/2022/AsyncFromSyncIteratorContinuation'),
	AsyncIteratorClose: require('es-abstract/2022/AsyncIteratorClose'),
	BigInt: require('es-abstract/2022/BigInt'),
	BigIntBitwiseOp: require('es-abstract/2022/BigIntBitwiseOp'),
	BinaryAnd: require('es-abstract/2022/BinaryAnd'),
	BinaryOr: require('es-abstract/2022/BinaryOr'),
	BinaryXor: require('es-abstract/2022/BinaryXor'),
	ByteListBitwiseOp: require('es-abstract/2022/ByteListBitwiseOp'),
	ByteListEqual: require('es-abstract/2022/ByteListEqual'),
	Call: require('es-abstract/2022/Call'),
	Canonicalize: require('es-abstract/2022/Canonicalize'),
	CanonicalNumericIndexString: require('es-abstract/2022/CanonicalNumericIndexString'),
	CharacterRange: require('es-abstract/2022/CharacterRange'),
	clamp: require('es-abstract/2022/clamp'),
	ClearKeptObjects: require('es-abstract/2022/ClearKeptObjects'),
	CloneArrayBuffer: require('es-abstract/2022/CloneArrayBuffer'),
	CodePointAt: require('es-abstract/2022/CodePointAt'),
	CodePointsToString: require('es-abstract/2022/CodePointsToString'),
	CompletePropertyDescriptor: require('es-abstract/2022/CompletePropertyDescriptor'),
	CompletionRecord: require('es-abstract/2022/CompletionRecord'),
	CopyDataProperties: require('es-abstract/2022/CopyDataProperties'),
	CreateAsyncFromSyncIterator: require('es-abstract/2022/CreateAsyncFromSyncIterator'),
	CreateDataProperty: require('es-abstract/2022/CreateDataProperty'),
	CreateDataPropertyOrThrow: require('es-abstract/2022/CreateDataPropertyOrThrow'),
	CreateHTML: require('es-abstract/2022/CreateHTML'),
	CreateIterResultObject: require('es-abstract/2022/CreateIterResultObject'),
	CreateListFromArrayLike: require('es-abstract/2022/CreateListFromArrayLike'),
	CreateMethodProperty: require('es-abstract/2022/CreateMethodProperty'),
	CreateNonEnumerableDataPropertyOrThrow: require('es-abstract/2022/CreateNonEnumerableDataPropertyOrThrow'),
	CreateRegExpStringIterator: require('es-abstract/2022/CreateRegExpStringIterator'),
	DateFromTime: require('es-abstract/2022/DateFromTime'),
	DateString: require('es-abstract/2022/DateString'),
	Day: require('es-abstract/2022/Day'),
	DayFromYear: require('es-abstract/2022/DayFromYear'),
	DaysInYear: require('es-abstract/2022/DaysInYear'),
	DayWithinYear: require('es-abstract/2022/DayWithinYear'),
	DefinePropertyOrThrow: require('es-abstract/2022/DefinePropertyOrThrow'),
	DeletePropertyOrThrow: require('es-abstract/2022/DeletePropertyOrThrow'),
	DetachArrayBuffer: require('es-abstract/2022/DetachArrayBuffer'),
	EnumerableOwnPropertyNames: require('es-abstract/2022/EnumerableOwnPropertyNames'),
	FlattenIntoArray: require('es-abstract/2022/FlattenIntoArray'),
	floor: require('es-abstract/2022/floor'),
	FromPropertyDescriptor: require('es-abstract/2022/FromPropertyDescriptor'),
	Get: require('es-abstract/2022/Get'),
	GetGlobalObject: require('es-abstract/2022/GetGlobalObject'),
	GetIterator: require('es-abstract/2022/GetIterator'),
	GetMatchIndexPair: require('es-abstract/2022/GetMatchIndexPair'),
	GetMatchString: require('es-abstract/2022/GetMatchString'),
	GetMethod: require('es-abstract/2022/GetMethod'),
	GetOwnPropertyKeys: require('es-abstract/2022/GetOwnPropertyKeys'),
	GetPromiseResolve: require('es-abstract/2022/GetPromiseResolve'),
	GetPrototypeFromConstructor: require('es-abstract/2022/GetPrototypeFromConstructor'),
	GetStringIndex: require('es-abstract/2022/GetStringIndex'),
	GetSubstitution: require('es-abstract/2022/GetSubstitution'),
	GetV: require('es-abstract/2022/GetV'),
	GetValueFromBuffer: require('es-abstract/2022/GetValueFromBuffer'),
	HasOwnProperty: require('es-abstract/2022/HasOwnProperty'),
	HasProperty: require('es-abstract/2022/HasProperty'),
	HourFromTime: require('es-abstract/2022/HourFromTime'),
	InLeapYear: require('es-abstract/2022/InLeapYear'),
	InstallErrorCause: require('es-abstract/2022/InstallErrorCause'),
	InstanceofOperator: require('es-abstract/2022/InstanceofOperator'),
	Invoke: require('es-abstract/2022/Invoke'),
	IsAccessorDescriptor: require('es-abstract/2022/IsAccessorDescriptor'),
	IsArray: require('es-abstract/2022/IsArray'),
	IsBigIntElementType: require('es-abstract/2022/IsBigIntElementType'),
	IsCallable: require('es-abstract/2022/IsCallable'),
	IsCompatiblePropertyDescriptor: require('es-abstract/2022/IsCompatiblePropertyDescriptor'),
	IsConcatSpreadable: require('es-abstract/2022/IsConcatSpreadable'),
	IsConstructor: require('es-abstract/2022/IsConstructor'),
	IsDataDescriptor: require('es-abstract/2022/IsDataDescriptor'),
	IsDetachedBuffer: require('es-abstract/2022/IsDetachedBuffer'),
	IsExtensible: require('es-abstract/2022/IsExtensible'),
	IsGenericDescriptor: require('es-abstract/2022/IsGenericDescriptor'),
	IsIntegralNumber: require('es-abstract/2022/IsIntegralNumber'),
	IsLessThan: require('es-abstract/2022/IsLessThan'),
	IsLooselyEqual: require('es-abstract/2022/IsLooselyEqual'),
	IsNoTearConfiguration: require('es-abstract/2022/IsNoTearConfiguration'),
	IsPromise: require('es-abstract/2022/IsPromise'),
	IsPropertyKey: require('es-abstract/2022/IsPropertyKey'),
	IsRegExp: require('es-abstract/2022/IsRegExp'),
	IsSharedArrayBuffer: require('es-abstract/2022/IsSharedArrayBuffer'),
	IsStrictlyEqual: require('es-abstract/2022/IsStrictlyEqual'),
	IsStringPrefix: require('es-abstract/2022/IsStringPrefix'),
	IsStringWellFormedUnicode: require('es-abstract/2022/IsStringWellFormedUnicode'),
	IsUnclampedIntegerElementType: require('es-abstract/2022/IsUnclampedIntegerElementType'),
	IsUnsignedElementType: require('es-abstract/2022/IsUnsignedElementType'),
	IsValidIntegerIndex: require('es-abstract/2022/IsValidIntegerIndex'),
	IsWordChar: require('es-abstract/2022/IsWordChar'),
	IterableToList: require('es-abstract/2022/IterableToList'),
	IteratorClose: require('es-abstract/2022/IteratorClose'),
	IteratorComplete: require('es-abstract/2022/IteratorComplete'),
	IteratorNext: require('es-abstract/2022/IteratorNext'),
	IteratorStep: require('es-abstract/2022/IteratorStep'),
	IteratorValue: require('es-abstract/2022/IteratorValue'),
	LengthOfArrayLike: require('es-abstract/2022/LengthOfArrayLike'),
	MakeDate: require('es-abstract/2022/MakeDate'),
	MakeDay: require('es-abstract/2022/MakeDay'),
	MakeMatchIndicesIndexPairArray: require('es-abstract/2022/MakeMatchIndicesIndexPairArray'),
	MakeTime: require('es-abstract/2022/MakeTime'),
	max: require('es-abstract/2022/max'),
	min: require('es-abstract/2022/min'),
	MinFromTime: require('es-abstract/2022/MinFromTime'),
	modulo: require('es-abstract/2022/modulo'),
	MonthFromTime: require('es-abstract/2022/MonthFromTime'),
	msFromTime: require('es-abstract/2022/msFromTime'),
	NewPromiseCapability: require('es-abstract/2022/NewPromiseCapability'),
	NormalCompletion: require('es-abstract/2022/NormalCompletion'),
	Number: require('es-abstract/2022/Number'),
	NumberBitwiseOp: require('es-abstract/2022/NumberBitwiseOp'),
	NumberToBigInt: require('es-abstract/2022/NumberToBigInt'),
	NumericToRawBytes: require('es-abstract/2022/NumericToRawBytes'),
	ObjectDefineProperties: require('es-abstract/2022/ObjectDefineProperties'),
	OrdinaryCreateFromConstructor: require('es-abstract/2022/OrdinaryCreateFromConstructor'),
	OrdinaryDefineOwnProperty: require('es-abstract/2022/OrdinaryDefineOwnProperty'),
	OrdinaryGetOwnProperty: require('es-abstract/2022/OrdinaryGetOwnProperty'),
	OrdinaryGetPrototypeOf: require('es-abstract/2022/OrdinaryGetPrototypeOf'),
	OrdinaryHasInstance: require('es-abstract/2022/OrdinaryHasInstance'),
	OrdinaryHasProperty: require('es-abstract/2022/OrdinaryHasProperty'),
	OrdinaryObjectCreate: require('es-abstract/2022/OrdinaryObjectCreate'),
	OrdinarySetPrototypeOf: require('es-abstract/2022/OrdinarySetPrototypeOf'),
	OrdinaryToPrimitive: require('es-abstract/2022/OrdinaryToPrimitive'),
	PromiseResolve: require('es-abstract/2022/PromiseResolve'),
	QuoteJSONString: require('es-abstract/2022/QuoteJSONString'),
	RawBytesToNumeric: require('es-abstract/2022/RawBytesToNumeric'),
	RegExpCreate: require('es-abstract/2022/RegExpCreate'),
	RegExpExec: require('es-abstract/2022/RegExpExec'),
	RegExpHasFlag: require('es-abstract/2022/RegExpHasFlag'),
	RequireObjectCoercible: require('es-abstract/2022/RequireObjectCoercible'),
	SameValue: require('es-abstract/2022/SameValue'),
	SameValueNonNumeric: require('es-abstract/2022/SameValueNonNumeric'),
	SameValueZero: require('es-abstract/2022/SameValueZero'),
	SecFromTime: require('es-abstract/2022/SecFromTime'),
	Set: require('es-abstract/2022/Set'),
	SetFunctionLength: require('es-abstract/2022/SetFunctionLength'),
	SetFunctionName: require('es-abstract/2022/SetFunctionName'),
	SetIntegrityLevel: require('es-abstract/2022/SetIntegrityLevel'),
	SetTypedArrayFromArrayLike: require('es-abstract/2022/SetTypedArrayFromArrayLike'),
	SetTypedArrayFromTypedArray: require('es-abstract/2022/SetTypedArrayFromTypedArray'),
	SetValueInBuffer: require('es-abstract/2022/SetValueInBuffer'),
	SortIndexedProperties: require('es-abstract/2022/SortIndexedProperties'),
	SpeciesConstructor: require('es-abstract/2022/SpeciesConstructor'),
	StringCreate: require('es-abstract/2022/StringCreate'),
	StringGetOwnProperty: require('es-abstract/2022/StringGetOwnProperty'),
	StringIndexOf: require('es-abstract/2022/StringIndexOf'),
	StringPad: require('es-abstract/2022/StringPad'),
	StringToBigInt: require('es-abstract/2022/StringToBigInt'),
	StringToCodePoints: require('es-abstract/2022/StringToCodePoints'),
	StringToNumber: require('es-abstract/2022/StringToNumber'),
	substring: require('es-abstract/2022/substring'),
	SymbolDescriptiveString: require('es-abstract/2022/SymbolDescriptiveString'),
	TestIntegrityLevel: require('es-abstract/2022/TestIntegrityLevel'),
	thisBigIntValue: require('es-abstract/2022/thisBigIntValue'),
	thisBooleanValue: require('es-abstract/2022/thisBooleanValue'),
	thisNumberValue: require('es-abstract/2022/thisNumberValue'),
	thisStringValue: require('es-abstract/2022/thisStringValue'),
	thisSymbolValue: require('es-abstract/2022/thisSymbolValue'),
	thisTimeValue: require('es-abstract/2022/thisTimeValue'),
	ThrowCompletion: require('es-abstract/2022/ThrowCompletion'),
	TimeClip: require('es-abstract/2022/TimeClip'),
	TimeFromYear: require('es-abstract/2022/TimeFromYear'),
	TimeString: require('es-abstract/2022/TimeString'),
	TimeWithinDay: require('es-abstract/2022/TimeWithinDay'),
	ToBigInt: require('es-abstract/2022/ToBigInt'),
	ToBigInt64: require('es-abstract/2022/ToBigInt64'),
	ToBigUint64: require('es-abstract/2022/ToBigUint64'),
	ToBoolean: require('es-abstract/2022/ToBoolean'),
	ToDateString: require('es-abstract/2022/ToDateString'),
	ToIndex: require('es-abstract/2022/ToIndex'),
	ToInt16: require('es-abstract/2022/ToInt16'),
	ToInt32: require('es-abstract/2022/ToInt32'),
	ToInt8: require('es-abstract/2022/ToInt8'),
	ToIntegerOrInfinity: require('es-abstract/2022/ToIntegerOrInfinity'),
	ToLength: require('es-abstract/2022/ToLength'),
	ToNumber: require('es-abstract/2022/ToNumber'),
	ToNumeric: require('es-abstract/2022/ToNumeric'),
	ToObject: require('es-abstract/2022/ToObject'),
	ToPrimitive: require('es-abstract/2022/ToPrimitive'),
	ToPropertyDescriptor: require('es-abstract/2022/ToPropertyDescriptor'),
	ToPropertyKey: require('es-abstract/2022/ToPropertyKey'),
	ToString: require('es-abstract/2022/ToString'),
	ToUint16: require('es-abstract/2022/ToUint16'),
	ToUint32: require('es-abstract/2022/ToUint32'),
	ToUint8: require('es-abstract/2022/ToUint8'),
	ToUint8Clamp: require('es-abstract/2022/ToUint8Clamp'),
	ToZeroPaddedDecimalString: require('es-abstract/2022/ToZeroPaddedDecimalString'),
	TrimString: require('es-abstract/2022/TrimString'),
	Type: require('es-abstract/2022/Type'),
	TypedArrayCreate: require('es-abstract/2022/TypedArrayCreate'),
	TypedArrayElementSize: require('es-abstract/2022/TypedArrayElementSize'),
	TypedArrayElementType: require('es-abstract/2022/TypedArrayElementType'),
	TypedArraySpeciesCreate: require('es-abstract/2022/TypedArraySpeciesCreate'),
	UnicodeEscape: require('es-abstract/2022/UnicodeEscape'),
	UTF16EncodeCodePoint: require('es-abstract/2022/UTF16EncodeCodePoint'),
	UTF16SurrogatePairToCodePoint: require('es-abstract/2022/UTF16SurrogatePairToCodePoint'),
	ValidateAndApplyPropertyDescriptor: require('es-abstract/2022/ValidateAndApplyPropertyDescriptor'),
	ValidateAtomicAccess: require('es-abstract/2022/ValidateAtomicAccess'),
	ValidateIntegerTypedArray: require('es-abstract/2022/ValidateIntegerTypedArray'),
	ValidateTypedArray: require('es-abstract/2022/ValidateTypedArray'),
	WeakRefDeref: require('es-abstract/2022/WeakRefDeref'),
	WeekDay: require('es-abstract/2022/WeekDay'),
	WordCharacters: require('es-abstract/2022/WordCharacters'),
	YearFromTime: require('es-abstract/2022/YearFromTime')
};

module.exports = ES2022;
